src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_Warnings.cs

```diff
@@ -859,6 +859,39 @@ private bool CheckForAssignmentToSelf(BoundAssignmentOperator node)
             return false;
         }
 
+        private void CheckForDeconstructionAssignmentToSelf(BoundTupleLiteral leftTuple, BoundExpression right)
+        {
+            while (right.Kind == BoundKind.Conversion)
+            {
+                right = ((BoundConversion)right).Operand;
```

Comment:
Yes, the first test has a `Deconstruction` conversion wrapping an `ImplicitTupleLiteral` conversion wrapping a `BoundConvertedTupleLiteral`.